packages/web-components/fast-element/src/templating/focusgroup.ts

```diff
@@ -0,0 +1,428 @@
+import {
+    keyArrowDown,
+    keyArrowLeft,
+    keyArrowRight,
+    keyArrowUp,
+    keyEnd,
+    keyHome,
+    keyPageDown,
+    keyPageUp,
+    keyTab,
+} from "@microsoft/fast-web-utilities";
+import { AttachedBehaviorHTMLDirective, ExecutionContext } from "..";
+import type { Behavior } from "../observation/behavior";
+import type { CaptureType } from "./template";
+
+/**
+ * The options used to configure focusgroups
+ * @public
+ */
+export interface FocusgroupBehaviorOptions {
+    wrap?: "both" | "horizontal" | "none" | "vertical";
+    direction?: "both" | "horizontal" | "vertical";
+    extend?: boolean;
+    bubble?: boolean;
+}
+
+export type DirectionGuide = {
+    up: {
+        index: number;
+        overlap?: number;
+        offset?: number;
+    };
+    right: {
+        index: number;
+        overlap?: number;
+        offset?: number;
+    };
+    down: {
+        index: number;
+        overlap?: number;
+        offset?: number;
+    };
+    left: {
+        index: number;
+        overlap?: number;
+        offset?: number;
+    };
+    top: {
+        index: number;
+        overlap?: number;
+        offset?: number;
+    };
+    bottom: {
+        index: number;
+        overlap?: number;
+        offset?: number;
+    };
+};
+
+/**
+ * Focusgroup behavior
+ * Handles keyboard arrowing inside a focusgroup
+ * @public
+ */
+export class FocusgroupBehavior implements Behavior {
+    private source: any;
+
+    public constructor(
+        private target: HTMLElement,
+        private options: FocusgroupBehaviorOptions
+    ) {
+        this.options = Object.assign(
+            {},
+            {
+                wrap: "both",
+                direction: "both",
+                extend: true,
+                bubble: false,
+            },
+            options
+        );
+        this.target = target;
+    }
+
+    private focusItems: HTMLElement[];
+
+    private positions: DirectionGuide[];
+
+    private firstElement: {
+        node: HTMLElement;
+        fromStart: number;
+        index: number;
+    };
+
+    private lastElement: {
+        node: HTMLElement;
+        fromEnd: number;
+        index: number;
+    };
+
+    /**
+     * Gets the elements under the focusgroup that are focusable
+     * @param node - root node to find focusable items under
+     * @returns - an array of html elements that are focusable to arrow between
+     * @public
+     */
+    public getFocusItems(node: HTMLElement = this.target): HTMLElement[] {
```

Comment:
Should we be using tabbable here like we do in dialog?

Reply:
Thanks! I was using a hard coded list that I wasn't happy with and did not want to maintain. I've copied your code from dialog for identifying tabbable elements.